### ISSUE TITLE
Embedded JS inside AcroForm button does not work as expected

### ISSUE THREAD

**Issue by pcaldeira-md** [mozilla/pdf.js#13391]
Attach (recommended) or Link to PDF file here: [unmodified-example.pdf](https://github.com/mozilla/pdf.js/files/6494921/unmodified-example.pdf)

Configuration:
- Web browser and its version: Firefox 88.0.1 (64-bit)
- Operating system and its version: macOS 11.3
- PDF.js version: unknown
- Is a browser extension: no

Steps to reproduce the problem:
1. Load the linked PDF in the browser and view it with pdf.js.
2. Edit the `Toto` text field.
3. Click the Ready button

What is the expected behavior? ![Screenshot 2021-05-17 at 17 48 57](https://user-images.githubusercontent.com/79838903/118518365-56f6d280-b738-11eb-89d5-6822207dfbeb.png)
The text field should become read-only (no editing possible) and the `Reset` and `Ready` buttons should disappear. This is behaviour I get when doing the above steps on Acrobat Reader.

What went wrong? ![Screenshot 2021-05-17 at 17 48 16](https://user-images.githubusercontent.com/79838903/118518220-329af600-b738-11eb-81be-33cf0ef837e8.png)

The text field remains editable and the buttons do not disappear.

The embedded JavaScript inside the button is as follows:
```javascript
for (var i = 0 ; i < this.numFields ; i++) {
  var field = this.getField(this.getNthFieldName(i));
  if (field.type == 'button' && (field.name == 'ready' || field.name == 'reset')) {
      field.hidden = true;
  } else if (field.type != 'button') {
      field.readonly = true;
  }
}
```

**Comment by calixteman** [comment]
The readonly issue is fixed in master (thx to https://github.com/mozilla/pdf.js/commit/17e9cfcd2a86bf6b72faf0ac3ac96fe4224731ba) but there is still an issue about buttons.
For now buttons are just drawn on the canvas and a clickable section element is added on top of it, so hidding a button means redraw all the canvas...
We should probably replace it by a real html button we can play with but I don't know if having a real button could lead to undesired side effects in other situations.
@timvandermeij what do you think about replacing the section by a real button ?

**Comment by Snuffleupagus** [comment]
> We should probably replace it by a real html button we can play with but I don't know if having a real button could lead to undesired side effects in other situations.

For one thing, *if* that's done it would need to be limited to only happen with `enableScripting = true` set such that we don't break the "normal" mode (since we cannot expect that everyone wants to enable scripting).

Furthermore there's also the question of what to do abou the appearanceStream. Can we simply use a regular HTML button, and skip the appearanceStream, or do we *somehow* need to try and use the "correct" appearance?

**Comment by calixteman** [comment]
Yeah my concern is mainly about appearanceStream and I've the same question as yours.
TBH, I'm pretty sure that's a bad idea to replace by a real button.
So maybe we could just provide a canvas to render the button and then put this canvas in the section, this way we could at least hide/show the button.
For now, `buttonSetIcon` and `buttonGetIcon` are not implemented because implementing them would imply to redraw the canvas but I think that it should be possible to do if we've a canvas by button (the "icon" finally).

**Comment by pcaldeira-md** [comment]
Thank you for having taken the time to look into this issue.

**Comment by Snuffleupagus** [comment]
WFM, using PDF.js `3.0.252` [987062c30] in Firefox Nightly 108 on Windows; perhaps fixed by PR #14247?
